cmd/integration_test.go

```diff
@@ -356,77 +357,68 @@ func TestSmokescreenIntegration(t *testing.T) {
 					}
 
 					// Report and open modes should always allow requests.
-					if action != acl.Enforce {
+					if policy != acl.Enforce {
 						expectAllow = true
 					}
 
-					targetScheme := "http"
-					if overTls {
-						targetScheme = "https"
+					testCase := &TestCase{
+						ExpectAllow: expectAllow,
+						Action:      policy,
+						OverTLS:     overTLS,
+						OverConnect: overConnect,
+						ProxyURL:    proxyServers[overTLS].URL,
+						TargetURL:   proxyTarget,
+						RoleName:    generateRoleForPolicy(policy),
 					}
 
-					testCase := &TestCase{
-						ExpectAllow:  expectAllow,
-						Action:       action,
-						OverTls:      overTls,
-						OverConnect:  overConnect,
-						ProxyURL:     servers[overTls].URL,
-						TargetPort:   port,
-						TargetHost:   host,
-						TargetScheme: targetScheme,
-						RoleName:     generateRoleForAction(action),
+					if expectAllow {
+						testCase.ExpectStatus = http.StatusOK
+						if overTLS && !authorizedHost {
+							// The Stripe API returns a 403 to a bare HTTP GET request
```

Comment:
Nice, this is a great compromise — we _do_ control this condition, or at least we can poke someone who does!